packages/dds/tree/src/core/edit-manager/editManager.ts

```diff
@@ -287,16 +290,76 @@ export class EditManager<
 		return this.changeFamily.intoDelta(this.rebaseLocalBranchOverTrunk());
 	}
 
-	public addLocalChange(revision: RevisionTag, change: TChangeset): Delta.Root {
+	public addLocalChange(
+		revision: RevisionTag,
+		change: TChangeset,
+		rebaseAnchors = true,
```

Comment:
This (`rebaseAnchors`) is clumsy, but I'm not sure how else to do it. `addLocalChange()` is called by `SharedTreeCore.applyChange()`, which is called in a few different contexts. When called by an `EditBuilder`, we don't want to rebase here because the edit builder does the rebasing already. In practice, it's possible to rebase the anchors everywhere we need to outside of this function. So, one path forward is to not have `addLocalChange` do any rebasing, and we can remove those checks from the edit manager unit tests. But then we should probably also rip the anchor rebasing out of `addSequencedChange` and `rollbackLocalChanges` too.

Reply:
For now, I'll leave it as is. I made a note in a future task to investigate this more.